pyfarm/master/api/jobgroups.py

```diff
@@ -0,0 +1,355 @@
+
+            {
+                "main_jobtype": "VARCHAR(64)",
+                "title": "VARCHAR(255)",
+                "user": "VARCHAR(255)",
+                "id": "INTEGER"
+            }
+
+    :statuscode 200: no error
+    """
+    schema_dict = JobGroup.to_schema()
+    # In the database, we are storing the user by id, but over the wire, we are
+    # using the username to identify the user instead.
+    schema_dict["user"] = "VARCHAR(%s)" % MAX_USERNAME_LENGTH
+    del schema_dict["user_id"]
+    schema_dict["main_jobtype"] = "VARCHAR(%s)" % MAX_JOBTYPE_LENGTH
+    del schema_dict["main_jobtype_id"]
+    return jsonify(schema_dict), OK
+
+class JobGroupIndexAPI(MethodView):
+    @validate_with_model(JobGroup, ignore=["user", "main_jobtype"],
+                         disallow=["user_id", "main_jobtype_id"])
+    def post(self):
+        """
+        A ``POST`` to this endpoint will create a new job group.
+
+        .. http:post:: /api/v1/jobgroups/ HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                POST /api/v1/jobgroups/ HTTP/1.1
+                Accept: application/json
+
+                {
+                    "title": "Test Group",
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType"
+                }
+
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 201 CREATED
+                Content-Type: application/json
+
+                {
+                    "id": 2,
+                    "jobs": [],
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType",
+                    "title": "Test Group"
+                }
+
+        :statuscode 201: a new job group was created
+        :statuscode 400: there was something wrong with the request (such as
+                         invalid columns being included)
+        """
+        username = g.json.pop("user")
+        user = User.query.filter_by(username=username).first()
+        if not user and AUTOCREATE_USERS:
+            user = User(username=username)
+            if AUTO_USERS_DEFAULT_DOMAIN:
+                user.email = username + "@" + AUTO_USERS_DEFAULT_DOMAIN
+            db.session.add(user)
+            logger.warning("User %s was autocreated on job group create",
+                           username)
+        elif not user:
+            return (jsonify(
+                error="User %s not found" % username), NOT_FOUND)
+
+        jobtype_name = g.json.pop("main_jobtype")
+        jobtype = JobType.query.filter_by(name=jobtype_name).first()
+        if not jobtype:
+            return (jsonify(
+                error="Jobtype %s not found" % jobtype_name), NOT_FOUND)
+
+        jobgroup = JobGroup(user=user, main_jobtype=jobtype, **g.json)
+        db.session.add(jobgroup)
+        db.session.commit()
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
```

Comment:
Same function but won't cause the whole request to fail if `jobgroup_data` does not contain the keys (which also means less work if we update the data later on):

```python
jobgroup_data.pop("user_id', None)
jobgroup_data.pop("main_jobtype_id", None)
```